sql/expression/unresolved.go

```diff
@@ -86,6 +86,103 @@ func (uc *UnresolvedColumn) WithChildren(children ...sql.Expression) (sql.Expres
 	return uc, nil
 }
 
+var _ sql.TableFunction = (*UnresolvedTableFunction)(nil)
+
+// UnresolvedTableFunction represents a table function that is not yet resolved.
+// This is a placeholder node, so methods such as Schema, RowIter, etc, are not
+// intended to be used.
+type UnresolvedTableFunction struct {
+	name      string
+	Arguments []sql.Expression
+	database  sql.Database
+}
+
+// NewUnresolvedTableFunction creates a new UnresolvedTableFunction node for a sql plan.
+func NewUnresolvedTableFunction(name string, arguments []sql.Expression) *UnresolvedTableFunction {
+	return &UnresolvedTableFunction{
+		name:      name,
+		Arguments: arguments,
+	}
+}
+
+// Database implements the Databaser interface
+func (utf UnresolvedTableFunction) Database() sql.Database {
+	return utf.database
+}
+
+// WithDatabase implements the Databaser interface
+func (utf UnresolvedTableFunction) WithDatabase(database sql.Database) (sql.Node, error) {
+	utf.database = database
+	return utf, nil
+}
+
+// TableFunctionName implements the TableFunction interface
+func (utf UnresolvedTableFunction) TableFunctionName() string {
+	return utf.name
+}
+
+// Description implements the TableFunction interface
+func (utf UnresolvedTableFunction) Description() string {
+	return fmt.Sprintf("unresolved table function %q", utf.TableFunctionName())
+}
+
+// Expressions implements the Expressioner interface
+func (utf UnresolvedTableFunction) Expressions() []sql.Expression {
+	return utf.Arguments
+}
+
+// WithExpressions implements the Expressioner interface
+func (utf UnresolvedTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
+	utf.Arguments = make([]sql.Expression, len(expression))
```

Comment:
Should do a length check on the arguments and return an error if they vary from what's already in the function.